src/test/java/com/cst438/controller/AssignmentControllerUnitTest.java

```diff
@@ -0,0 +1,157 @@
+
+            // Create user for gradeAssignmentSuccess
+            UserDTO student1 = new UserDTO(0, "Student One", "student1@example.com", "STUDENT");
+            mvc.perform(
+                            MockMvcRequestBuilders
+                                    .post("/users")
+                                    .contentType(MediaType.APPLICATION_JSON)
+                                    .content(asJsonString(student1)))
+                    .andReturn()
+                    .getResponse();
+
+            // Create user for gradeAssignmentInvalidAssignmentId
+            UserDTO student2 = new UserDTO(0, "Student Two", "student2@example.com", "STUDENT");
+            mvc.perform(
+                            MockMvcRequestBuilders
+                                    .post("/users")
+                                    .contentType(MediaType.APPLICATION_JSON)
+                                    .content(asJsonString(student2)))
+                    .andReturn()
+                    .getResponse();
+
+            // Create user for enterFinalGradesSuccess
+            UserDTO student3 = new UserDTO(0, "Student Three", "student3@example.com", "STUDENT");
+            mvc.perform(
+                            MockMvcRequestBuilders
+                                    .post("/users")
+                                    .contentType(MediaType.APPLICATION_JSON)
+                                    .content(asJsonString(student3)))
+                    .andReturn()
+                    .getResponse();
+
+            setUpIsDone = true;
+        }
+    }
+
+    @Test
+    public void gradeAssignmentSuccess() throws Exception {
+        MockHttpServletResponse response = mvc.perform(
+                        MockMvcRequestBuilders
+                                .get("/assignments/1/grades")
+                                .contentType(MediaType.APPLICATION_JSON))
+                .andReturn()
+                .getResponse();
+
+        assertEquals(200, response.getStatus(), "Expected status 200 but got " + response.getStatus());
+
+        GradeDTO[] grades = fromJsonString(response.getContentAsString(), GradeDTO[].class);
+        GradeDTO[] updatedGrades = new GradeDTO[grades.length];
+        for (int i = 0; i < grades.length; i++) {
+            GradeDTO grade = grades[i];
+            updatedGrades[i] = new GradeDTO(grade.gradeId(), grade.studentName(), grade.studentEmail(), grade.assignmentTitle(), grade.courseId(), grade.sectionId(), grade.score() + 5); // Example modification
+        }
+
+        response = mvc.perform(
+                        MockMvcRequestBuilders
+                                .put("/grades")
+                                .content(asJsonString(updatedGrades))
+                                .contentType(MediaType.APPLICATION_JSON))
+                .andReturn()
+                .getResponse();
+
+        assertEquals(200, response.getStatus(), "Expected status 200 but got " + response.getStatus());
```

Comment:
This test looks like it meets all of the requirements:

- GET /assignments/{assignmentId}/grades is invoked
- GradeDTO objects have updated scores & are saved using a PUT call to /grades
- Both requests have an assert statement for the status code

Reply:
Ideally I think we would want to get the grades back again to verify that they have been updated properly, but it says nothing about doing this in the rubric.